packages/slo-stat-bot/src/slo-appliesTo.ts

```diff
@@ -0,0 +1,210 @@
+ * @param githubLabels is slo rule for github labels that must exist in issue
+ * @returns true if githubLabels applies to issues else false
+ */
+export const isValidGithubLabels = async function isValidGithubLabels(
+  issueLabels: string[],
+  githubLabels: string | string[] | undefined
+): Promise<boolean> {
+  if (!githubLabels) {
+    return true;
+  }
+
+  githubLabels = await convertToArray(githubLabels);
+  githubLabels.forEach((label: string) => label.toLowerCase());
+  const isSubSet = githubLabels.every((label: string) =>
+    issueLabels.includes(label)
+  );
+  return isSubSet;
+};
+
+/**
+ * Function checks if all the excluded github labels is not in issue labels
+ * @param issueLabels of the issue
+ * @param excludedGitHubLabels is slo rule for excluded github labels that must exist in issue
+ * @returns true if excludedGitHubLabels applies to issues else false
+ */
+export const isValidExcludedLabels = async function isValidExcludedLabels(
+  issueLabels: string[],
+  excludedGitHubLabels: string | string[] | undefined
+): Promise<boolean> {
+  if (!excludedGitHubLabels) {
+    return true;
+  }
+
+  excludedGitHubLabels = await convertToArray(excludedGitHubLabels);
+  excludedGitHubLabels.forEach((label: string) => label.toLowerCase());
+  const isElementExist = excludedGitHubLabels.some((label: string) =>
+    issueLabels.includes(label)
+  );
+  return !isElementExist;
+};
+
+/**
+ * Function checks if the rule (priority or type) exists in issue labels
+ * @param issueLabels of the issue
+ * @param rule is either priority or type (ex: bug, enhancement) of issue
+ * @param title of the rule
+ * @returns true if rule applies to issue else false
+ */
+export const isValidRule = async function isValidRule(
+  issueLabels: string[],
+  rule: string | undefined,
+  title: string
+) {
+  if (!rule) {
+    return true;
+  }
+
+  rule = rule.toLowerCase();
+
+  if (issueLabels.includes(rule)) {
+    return true;
+  }
+  if (issueLabels.includes(title + rule)) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Function converts the variable to an array if it is a string
+ * @param variable can either be array or string
+ * @returns an array
+ */
+export const convertToArray = async function convertToArray(
+  variable: string[] | string
+): Promise<string[]> {
+  if (typeof variable === 'string') {
+    return [variable];
+  }
+  return variable;
+};
+
+/**
+ * Function determines if slo applies to the issue
+ * @param type specifies if event is issue or pr
+ * @param slo rules
+ * @param issueLabels of issue or pr
+ * @param number of issue or pr
+ * @returns true if slo applies to issue else false
+ */
+export const doesSloApply = async function doesSloApply(
+  type: string,
+  slo: SLORules,
+  issueLabels: string[] | null,
+  number: number
+): Promise<boolean> {
+  const sloString = JSON.stringify(slo, null, 4);
+
+  if (Object.keys(slo.appliesTo).length === 0) {
+    return true;
+  }
+
+  if (issueLabels === null) {
```

Comment:
Might be worth checking for falsy vs. null, I'm never quite sure what you get with these methods.

Reply:
I could just do check for `(!issueLabels)` which will consider both the case if it is null or falsy.